src/test/java/seedu/address/logic/OverdueCheckerTest.java

```diff
@@ -0,0 +1,99 @@
+package seedu.address.logic;
+
+import static junit.framework.TestCase.assertEquals;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Predicate;
+
+import javafx.application.Platform;
+import javafx.collections.ObservableList;
+import seedu.address.model.Model;
+import seedu.address.model.ReadOnlyDeskBoard;
+import seedu.address.model.activity.Activity;
+import seedu.address.model.activity.exceptions.ActivityNotFoundException;
+import seedu.address.model.activity.exceptions.DuplicateActivityException;
+import seedu.address.testutil.TypicalActivities;
+
+public class OverdueCheckerTest {
+
+    private ScheduledExecutorService executor = Executors.newScheduledThreadPool(1);
+
+    //TODO
+    /**
+     *
+     */
```

Comment:
Remember to fill up this comment, if there is no comments needed then it should be removed =).